plugin.lua

```diff
@@ -864,43 +898,168 @@ local function show_tile_context_menu(ts, ti, folders, folder, indexInFolder)
+              local tilemapCel = layer:cel(usedFrame)
+              covertTilemapCelToCel(tilemapCel, tempSprite, newLayer, 1)
+            end
+            newLayer = tempSprite:newLayer()
+          end
+        end
+        app.command.FlattenLayers {}
+        app.activeLayer.name = "Faded Layer"
+        app.command.LayerOpacity {
+          opacity = 128
+        }
+        for i=1, #ts-2, 1 do
+          tempSprite:newEmptyFrame()
+        end
+        local frames = {}
+        for i=1, #tempSprite.frames, 1 do
+          table.insert(frames, i)
+        end
+        app.range.frames = frames
+        app.range.layers = { tempSprite.layers[1] }
+        app.command.LinkCels {}
+        app.activeLayer.isEditable = false
+
+        -- Add Attachment editing layer
+        editAttachmentLayer = tempSprite:newLayer()
+        editAttachmentLayer.name = originalLayer.name
+
+        -- Add Reference point layer
+        newLayer = tempSprite:newLayer()
+        newLayer.name = "Reference Point"
+        local pos = parentAnchorPos -
+          Point(refCrossImage.width/2, refCrossImage.height/2)
+        for i=1, #tempSprite.frames, 1 do
+          tempSprite:newCel(newLayer, i, refCrossImage, pos)
+        end
+        app.activeLayer.isEditable = false
+        app.activeLayer = editAttachmentLayer
+        app.activeFrame = ti
+
+        -- Filling the Attachment editing layer
+        for i=1, #tempSprite.frames, 1 do
+          local refPos
+          if refTileset:tile(i).properties(PK).ref then
+            refPos = refTileset:tile(i).properties(PK).ref
+          else
+            refPos = Point(tileSize.width/2, tileSize.height/2)
+          end
+          tempSprite:newCel(editAttachmentLayer,
+                            i,
+                            ts:tile(i).image,
+                            parentAnchorPos - refPos)
+          table.insert(attachmentOriginalPositions, parentAnchorPos - refPos)
+        end
+
+        -- Making the array of changes on tempSprite cels
+        changeDetectionOnCels = {}
+        for i=1, #tempSprite.frames, 1 do
+          table.insert(changeDetectionOnCels, false)
+        end
       end)
 
     local function accept()
+      local changes = changeDetectionOnCels
+      changeDetectionOnCels = nil
+      app.activeSprite = spr
       if tempSprite ~= nil then
-        local image = Image(ts:tile(ti).image.width, ts:tile(ti).image.height)
-        image:drawImage(app.activeCel.image, app.activeCel.position)
-        app.activeSprite = spr
-        app.transaction("Tile modified",
-          function()
-            ts:tile(ti).image = image
-          end)
+        for i=1, #tempSprite.frames, 1 do
+          if changes[i] and editAttachmentLayer:cel(i) then
+            local transparentColor = tempSprite.transparentColor
+            local celImage = editAttachmentLayer:cel(i).image
+            if not isEmptyImage(celImage, transparentColor) then
```

Comment:
Replace isEmptyImage() with built-in Image:isEmpty() function:
```suggestion
            local celImage = editAttachmentLayer:cel(i).image
            if not celImage:isEmpty() then
```